emcc.py

```diff
@@ -1280,6 +1282,10 @@ def check(input_file):
       shared.Settings.USE_LEGACY_DYNCALLS = 1
       shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$getDynCaller']
 
+      if shared.Settings.ASYNCIFY_ADVISE:
+        shared.Settings.DEBUG_LEVEL = max(shared.Settings.DEBUG_LEVEL, 2)
+        options.profiling = True
```

Comment:
I think instead of this `if`, we can set `intermediate_debug_info` lower down to `True` (which we do for a few other settings, including asyncify settings).

Reply:
At which point lower down?

Reply:
Sorry, just search for `intermediate_debug_info = ` - there's only one such variable in the whole file.

Reply:
Ah, gotcha, yes that's much neater.

Reply:
Except, it doesn't work. https://app.circleci.com/pipelines/github/emscripten-core/emscripten/7930/workflows/623f7a37-143d-4c23-9c25-667e65b7c43f/jobs/356184

Reply:
Very odd, I can't think why this wouldn't work. It should be doing the exact same thing for `ASYNCIFY_ADD` as for this option, they both need the Names section...

I'd take a look at `EMCC_DEBUG=1` output. You should see that we don't strip all debug info before running asyncify, and also you can inspect the temp files and see if debug info is there. It seems impossible that when using `ASYNCIFY_ADD` the names section is there but not when using `ASYNCIFY_ADVISE`, so any difference between two such builds is a hint to the bug I think.

Reply:
I needed to add a similar check to emscripten.py.